### ISSUE TITLE
rxjs-throw-error should not complain about throwing subtypes of Error

### ISSUE THREAD

**Issue by felixfbecker** [cartant/rxjs-tslint-rules#86]
```ts
throw Object.assign(new Error('Not Found'), { code: 'NOT_FOUND' })
```
currently creates a complaint, but shouldn't because the type of this exception is `Error & { code: string }`, which is a subtype of `Error`.

**Comment by cartant** [comment]
Note for self: the bug is most likely in the implementation of `couldBeType`, as `Object.assign` has this definition:

```ts
assign<T, U>(target: T, source: U): T & U;
```

**Comment by cartant** [comment]
This should be fixed in 4.17.1.
